content/en/docs/concepts/configuration/secret.md

```diff
@@ -106,7 +127,23 @@ system, without being directly exposed to the Pod. For example, Secrets can hold
 credentials that other parts of the system should use to interact with external
 systems on your behalf.
 
-### Using Secrets as files from a Pod
+{{< caution >}}
+Secret volume sources are validated to ensure that the specified object
+reference actually points to an object of type Secret. Therefore, a secret
+needs to be created before any Pods that depend on it.  
+
+If the secret cannot be fetched because it does not exist or
+because of a temporary lack of connection to the API server, the kubelet
+periodically retries running that Pod. The kubelet also reports an Event
+for that Pod, including details of the problem fetching the Secret.
+
+If your application can start without the secret data, and then adapt once
+it becomes available, you can mark the Secret as optional in the Pod specification.  
+None of the Pod's containers will start until all non-optional Secrets are
+available.
+{{< /caution >}}
```

Comment:
Please remove the 'caution' shortcode?

Reply:
> None of the Pod's containers will start until all non-optional Secrets are available.

Please revise this "none ... until all non-optional ..." sentence.

Reply:
I could switch this from a caution to a note.

Reply:
This is a bit of a mashup of note and warning.
You are indicating to the reader that Secrets can be declared "optional" in the Pod configuration. 
Then secondly, you warn that a Pod's containers are prevented from starting until the "required" or "non-optional" Secrets are ready. Could the first sentence move out of the callout?